src/synthesizer/photoionisation/cloudy23.py

```diff
@@ -237,77 +257,127 @@ def create_cloudy_input(
         """
 
         # If grain physics are required, add this self-consistently
-        if (abundances.dust_to_metal_ratio > 0) & params["grains"]:
-            delta_C = 10 ** abundances.total["C"] - 10 ** abundances.gas["C"]
-            delta_PAH = 0.01 * (10 ** abundances.total["C"])
-            delta_graphite = delta_C - delta_PAH
-            delta_Si = (
-                10 ** abundances.total["Si"] - 10 ** abundances.gas["Si"]
-            )
-            orion_C_abund = -3.6259
-            orion_Si_abund = -4.5547
-            PAH_abund = -4.446
-            f_graphite = delta_graphite / (10 ** (orion_C_abund))
-            f_Si = delta_Si / (10 ** (orion_Si_abund))
-            f_pah = delta_PAH / (10 ** (PAH_abund))
+        if params["grains"] is None:
+            f_graphite, f_Si, f_pah = 0, 0, 0
+
+        else:
+            if params["no_grain_scaling"] is False:
+                delta_C = (
+                    10 ** abundances.total["C"] - 10 ** abundances.gas["C"]
+                )
+                delta_PAH = 0.01 * (10 ** abundances.total["C"])
+                delta_graphite = delta_C - delta_PAH
+                delta_Si = (
+                    10 ** abundances.total["Si"] - 10 ** abundances.gas["Si"]
+                )
+
+                # define the reference abundances for the different grain types
+                # this should be the dust-phase abundance in the particular
+                # reference environment.
+                if params["grains"] == "Orion":
+                    reference_C_abund = -3.6259
+                    reference_Si_abund = -4.5547
+                # this is incorrect since the abundance in the ISM is probably
+                # different.
+                elif params["grains"] == "ISM":
+                    reference_C_abund = -3.6259
+                    reference_Si_abund = -4.5547
+                else:
+                    raise UnimplementedFunctionality(
+                        "Only Orion grains are currently implemented"
+                    )
+
+                PAH_abund = -4.446
+                f_graphite = delta_graphite / (10 ** (reference_C_abund))
+                f_Si = delta_Si / (10 ** (reference_Si_abund))
+                f_pah = delta_PAH / (10 ** (PAH_abund))
+
+            else:
+                f_graphite = 1.0
+                f_Si = 1.0
+                f_pah = 1.0
+
             command = (
-                f"grains Orion graphite {f_graphite} "
-                f"\ngrains Orion silicate {f_Si} \ngrains "
-                f"PAH {f_pah}"
+                f"grains {params['grains']} graphite {f_graphite} \n"
+                f"grains {params['grains']} silicate {f_Si} \n"
+                f"grains PAH {f_pah}"
             )
+
             cinput.append(command + "\n")
-        else:
-            f_graphite, f_Si, f_pah = 0, 0, 0
 
+    # NOTE FROM SW (01/03/24): WHILE THIS IS DESIRABLE I AM NOT CONVINCED IT
```

Comment:
I don't think leaving big blocks of commented code is a good idea. All of this exists within the git history so can be retrieved. If you don't want to go digging then make yourself a branch with this uncommented to work on in the future.

Reply:
Done